### PR TITLE
feat(chat): shared with me attachmeents e2e tests

### PR DESCRIPTION
**Pull request by nartovm**
**Description:**

shared with me attachmeents e2e tests

**UI changes**

<Please, provide Screenshots or Figma links>

**Checklist:**

- [ ] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [ ] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
Unknown release type "Feat" found in pull request title "Feat(chat): shared with me attachmeents e2e tests". 

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

File: apps/chat-e2e/src/assertions/downloadAssertion.ts [hunk 1]
```diff
@@ -12,6 +12,20 @@ export class DownloadAssertion {
     expect(downloadedData.path).toMatch(new RegExp(`${expectedExtension}$`));
   }
 
+  public async assertJsonFileIsDownloaded(downloadedData: UploadDownloadData) {
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/downloadAssertion.ts]: this method is identical to `assertFileIsDownloaded()`

File: apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts [hunk 2]
```diff
@@ -36,4 +37,41 @@ export class ManageAttachmentsAssertion {
       .soft(arrowIconColor[0], ExpectedMessages.sharedIconColorIsValid)
       .toBe(expectedColor);
   }
+  public async assertEntityState(
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts:46]: please, add line break before the method

File: apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts [hunk 3]
```diff
@@ -36,4 +37,41 @@ export class ManageAttachmentsAssertion {
...
+    entity: TreeEntity,
+    fileModalSection: FileModalSection,
+    expectedState: ElementState,
+  ) {
+    let entityTree: AttachFilesTree;
+    if (fileModalSection === FileModalSection.AllFiles) {
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts]: what do you think about using `switch-case` and adding all available `FileModalSection` ?

File: apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts [hunk 4]
```diff
@@ -36,4 +37,41 @@ export class ManageAttachmentsAssertion {
...
+          .soft(entityLocator, ExpectedMessages.entityIsVisible)
+          .toBeVisible()
+      : await expect
+          .soft(entityLocator, ExpectedMessages.entityIsNotVisible)
+          .toBeHidden();
+  }
+
+  public async assertSectionState(
+    section: FileModalSection,
+    state: 'expanded' | 'collapsed',
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts]: please refer to `ElementCaretState` type

File: apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts [hunk 5]
```diff
@@ -36,4 +37,41 @@ export class ManageAttachmentsAssertion {
...
+  ) {
+    const sectionElement = this.attachFilesModal.getSectionElement(section);
+    const isExpanded =
+      await this.attachFilesModal.isSectionExpanded(sectionElement);
+    const expectedIsExpanded = state === 'expanded'; // Clearer representation of the expected value
+    expect(isExpanded, `Section "${section}" is ${state}`).toBe(
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/manageAttachmentsAssertion.ts]: I guess the clearer option would be:

```
state === 'expanded'
      ? expect(isExpanded, `Section "${section}" is ${state}`).toBeTruthy()
      : expect(isExpanded, `Section "${section}" is ${state}`).toBeFalsy();
```

File: apps/chat-e2e/src/core/dialFixtures.ts [hunk 6]
```diff
@@ -730,7 +730,7 @@ const dialTest = test.extend<
     const conversationAssertion = new ConversationAssertion(conversations);
     await use(conversationAssertion);
   },
-  attachedFilesAssertion: async ({ attachFilesModal }, use) => {
+  manageAttachmentsAssertion: async ({ attachFilesModal }, use) => {
     const attachedFilesAssertion = new ManageAttachmentsAssertion(
```
Comment (irinakartun) [inline apps/chat-e2e/src/core/dialFixtures.ts]: rename const to `manageAttachmentsAssertion` as well

**Comment by nartovm** [comment]
//deploy-review

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/assertions/downloadAssertion.ts [hunk 7]
```diff
@@ -3,7 +3,19 @@ import { UploadDownloadData } from '@/src/ui/pages';
...
+}
+type FileReader = (path: string) => string | Buffer | object | undefined;
+
 export class DownloadAssertion {
+  private static fileReaders: Record<FileType, FileReader> = {
+    [FileType.JSON]: FileUtil.readJsonFileData,
+    [FileType.PLAIN]: FileUtil.readPlainFileData,
+    // [FileType.JPG]: FileUtil.readJpgFileData, //class can be extended to use with different file types
+  };
   public async assertDownloadFileExtension(
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/downloadAssertion.ts:20]: please, add line break before the method

File: apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts [hunk 8]
```diff
@@ -477,3 +487,530 @@ dialSharedWithMeTest(
...
+        );
+      conversationData.resetData();
+
+      await dataInjector.createConversations([
+        conversationWithTwoRequestsWithAttachments,
+        conversationWithTwoResponsesWithAttachments,
+        user1ConversationInFolder,
+      ]);
+
+      // await localStorageManager.setSelectedConversation(
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts]: remove if not needed

File: apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts [hunk 9]
```diff
@@ -477,3 +487,530 @@ dialSharedWithMeTest(
...
+        );
+        await additionalShareUserSendMessage.attachmentMenuTrigger.click();
+
+        await additionalShareUserAttachmentDropdownMenu.selectMenuOption(
+          UploadMenuOptions.attachUploadedFiles,
+        );
+      },
+    );
+
+    for (const section of [
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts:785]: can be replaced with:
`for (const section of Object.values(FileModalSection))`

File: apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts [hunk 10]
```diff
@@ -477,3 +487,530 @@ dialSharedWithMeTest(
...
+      FileModalSection.Organization,
+      FileModalSection.SharedWithMe,
+      FileModalSection.AllFiles,
+    ]) {
+      for (const state of ['collapsed', 'expanded']) {
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/sharedFilesAttachments.test.ts]: please, refer to `ElementCaretState` type

File: apps/chat-e2e/src/ui/webElements/attachFilesModal.ts [hunk 11]
```diff
@@ -67,6 +113,25 @@ export class AttachFilesModal extends BaseElement {
     return this.allFilesTree;
   }
 
+  public getSectionElement(section: FileModalSection): Locator {
+    switch (section) {
+      case FileModalSection.AllFiles:
+        return this.rootLocator.locator(
```
Comment (irinakartun) [inline apps/chat-e2e/src/ui/webElements/attachFilesModal.ts]: you may use `BaseElement` as a return type and transform the implementation like this:

`return this.getChildElementBySelector(AttachFilesModalSelectors.allFilesContainer,);`

File: apps/chat-e2e/src/utils/fileUtil.ts [hunk 12]
```diff
@@ -16,11 +16,15 @@ export class FileUtil {
     return filename;
   }
 
-  public static readFileData(path: string) {
+  public static readJsonFileData(path: string) {
```
Comment (irinakartun) [inline apps/chat-e2e/src/utils/fileUtil.ts:19]: please, add like break before the method

File: apps/chat-e2e/src/ui/webElements/search.ts [hunk 13]
```diff
@@ -17,10 +16,7 @@ export class Search extends BaseElement {
     }
     return this.filter;
   }
-
-  public searchInput = this.getChildElementBySelector(Tags.input);
-
   public async setSearchValue(value: string) {
```
Comment (irinakartun) [inline apps/chat-e2e/src/ui/webElements/search.ts]: please, add like break before the method

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by irinakartun — DISMISSED** [review]

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review


>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>- **Deployment failed**: [canceled](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/535799)

**Review by irinakartun — DISMISSED** [review]

**Comment by nartovm** [comment]
/deploy-review


>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by irinakartun — DISMISSED** [review]

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2518.nightly-test.deltixhub.io
>E2E tests status: failed
>E2E tests status: success

**Review by irinakartun — APPROVED** [review]

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**